server.go

```diff
@@ -302,14 +1037,12 @@ func (s *server) handleDonePeerMsg(state *peerState, p *peer) {
 		if e == p {
 			// Issue an asynchronous reconnect if the peer was a
 			// persistent outbound connection.
-			if !p.inbound && p.persistent && atomic.LoadInt32(&s.shutdown) == 0 {
-				delete(list, e)
-				e = newOutboundPeer(s, p.addr, true, p.retryCount+1)
-				list[e] = struct{}{}
+			if !p.Inbound() && p.Persistent() && atomic.LoadInt32(&s.shutdown) == 0 {
+				s.RetryPeer(p)
```

Comment:
I don't see anything in the `RetryPeer` path updating the peer in the `state.peers` or `state.outboundPeers` lists, so the check above to see if the peer is one of the ones that should be retried due to being persistent will never be triggered give the `if e == p` check isn't going to match since it's a new peer pointer after the retry.